source/common/upstream/upstream_impl.h

```diff
@@ -1111,6 +1119,7 @@ class ClusterInfoImpl : public ClusterInfo,
   UpstreamFactoryContextImpl upstream_context_;
   std::unique_ptr<envoy::config::cluster::v3::UpstreamConnectionOptions::HappyEyeballsConfig>
       happy_eyeballs_config_;
+  const std::unique_ptr<Envoy::Orca::LrsReportMetricNames> lrs_report_metric_names_;
```

Comment:
Where the value will be initialized? (if you want to do it in separated PR, I will suggest you to merge it here because I think it won't be too complex and could make this PR more integral.)

Reply:
Wow, great catch! The `upstream_impl.cc` got dropped from the commit when I've transferred it from internal repository to the GitHub. The `router_test` didn't catch it because it mocks the `ClusterInfo`.

I've added the `upstream_impl.cc` to the PR.